includes/class-coblocks-plugin-deactivation.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * Register plugin deactivation class.
+ *
+ * @package CoBlocks
+ */
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * CoBlocks_Plugin_Deactivation Class
+ *
+ * @since NEXT
+ */
+class CoBlocks_Plugin_Deactivation {
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+
+		add_action( 'admin_footer-plugins.php', array( $this, 'admin_coblocks_deactivation_modal' ) );
+
+		add_filter( 'admin_enqueue_scripts', array( $this, 'equeue_scripts' ) );
+
+	}
+
+	/**
+	 * Loads the asset file for the given script or style.
+	 * Returns a default if the asset file is not found.
+	 *
+	 * @param string $filepath The name of the file without the extension.
+	 *
+	 * @return array The asset file contents.
+	 */
+	public function get_asset_file( $filepath ) {
+
+		$asset_path = COBLOCKS_PLUGIN_DIR . $filepath . '.asset.php';
+
+		return file_exists( $asset_path )
+			? include $asset_path
+			: array(
+				'dependencies' => array(),
+				'version'      => COBLOCKS_VERSION,
+			);
+
+	}
+
+	/**
+	 * Register meta.
+	 */
+	public function equeue_scripts() {
+
+		global $pagenow;
+
+		if ( 'plugins.php' !== $pagenow ) {
+
+			return;
+
+		}
+
+		$name       = 'coblocks-plugin-deactivation';
+		$filepath   = 'dist/' . $name;
+		$asset_file = $this->get_asset_file( $filepath );
+
+		// Enqueue modal script.
+		wp_enqueue_script(
+			'coblocks-plugin-deactivation',
+			COBLOCKS_PLUGIN_URL . $filepath . '.js',
+			$asset_file['dependencies'],
+			$asset_file['version'],
+			true
+		);
+
+		$wpnux_export_data = json_decode( get_option( 'wpnux_export_data', '[]' ), true );
+
+		// Pass data.
+		wp_localize_script(
+			'coblocks-plugin-deactivation',
+			'coblocksDeactivateData',
+			array(
+				'hostname'        => gethostname(),
+				'domain'          => site_url(),
+				'coblocksVersion' => COBLOCKS_VERSION,
+				'wpVersion'       => $GLOBALS['wp_version'],
+				'wpOptions'       => array(
+					'persona' => isset( $wpnux_export_data['_meta']['persona'] ) ? $wpnux_export_data['_meta']['persona'] : null,
+					'skill'   => isset( $wpnux_export_data['_meta']['skill'] ) ? $wpnux_export_data['_meta']['skill'] : null,
+				),
+			)
+		);
+
+		// Styles.
+		$name       = 'style-coblocks-plugin-deactivation';
+		$filepath   = 'dist/' . $name;
+		$asset_file = $this->get_asset_file( $filepath );
```

Comment:
@sbalakrishnan1-godaddy I don't think this is stricly needed given we are only using the version number. We can use the version number of JS above here https://github.com/godaddy-wordpress/coblocks/pull/2266/files#diff-f4655db9906d8b47edf37aae513e2191a6e714255580a417ce1ca93d8eab9336R67.